lang/rust/rust.py

```diff
@@ -1,215 +1,522 @@
-from talon import Module, Context, actions, ui, imgui, settings
+from typing import Dict
+
+from talon import Context, Module, actions, settings
 
 mod = Module()
+# rust specific grammar
+mod.list('code_macros', desc='List of macros for active language')
+mod.list('code_trait', desc='List of traits for active language')
+
+
+@mod.capture(rule='{user.code_macros}')
+def code_macros(m) -> str:
+    """Returns a macro name"""
+    return m.code_macros
+
+
+@mod.action_class
+class Actions:
+    def code_operator_structure_dereference():
```

Comment:
If these actions are only going to be used in Rust, you don't really need to abstract them like this; might be simpler just to put their implementations into the rust.talon file. Not a big deal either way though.

Reply:
that's completely fair, I was just trying to keep these functions inside the python file, because their implementation is very similar to implementations of tags below. I'll bear this in mind for the future.